src/org/usfirst/frc4904/robot/subsystems/GearIO.java

```diff
@@ -0,0 +1,64 @@
+package org.usfirst.frc4904.robot.subsystems;
+
+
+import org.usfirst.frc4904.standard.commands.Idle;
+import org.usfirst.frc4904.standard.subsystems.motor.Motor;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+public class GearIO extends Subsystem {
+	protected final Motor intakeRoller;
+	protected final DoubleSolenoid gullWings;
+	protected final DoubleSolenoid ramp;
+	protected GearState currentState;
+
+	public GearIO(Motor intakeRoller, DoubleSolenoid gullWings, DoubleSolenoid ramp) {
+		this.intakeRoller = intakeRoller;
+		this.gullWings = gullWings;
+		this.ramp = ramp;
+		setState(GearState.INTAKE);
+	}
+
+	public static enum GearState {
```

Comment:
That was my issue - the ramp needs to be controlled independently from it. We put it out once at the beginning of the match, and only retract it for climbing.